### PR TITLE
Updated changelevel serverside implementation

### PR DESCRIPTION
**Pull request by Kefta**
This now matches GMod's menu-getmaps' categories with a few more optimizations:
- CS:GO, Portal 2, Dino D-Day, L4D2, and Alien Swarm are no longer shown as they can't be displayed
- Map categories are now put into two tables: mapNames/ignoreNames and mapPatterns/ignorePatterns to only pattern search when necessary
-  ipairs loops :)

### CODE REVIEW

**Comment by Bo98** [comment]
> * No more useless goto call that did the same thing as the continue keyword

It wasn't useless. It continued the parent for loop. Right now you have created a dead for loop.

File: gamemode/modules/fadmin/fadmin/changelevel/sv_init.lua [hunk 1]
```diff
@@ -4,61 +4,98 @@ local function ChangeLevel(ply, cmd, args)
...
-                    table.insert(mapList[cat], name)
-                else
-                    category = "CS: Global Offensive"
-                end
-            end
+
+        -- Throw all uncategorized maps into Other
+        mapCategory = mapCategory or "Other"
+        -- Don't show CS:GO maps
+        if mapCategory == "Counter-Strike" and file.Exists("maps/" .. name .. ".bsp", "csgo") then
```
Comment (Bo98) [inline gamemode/modules/fadmin/fadmin/changelevel/sv_init.lua]: How will this behave for maps that exists in both CS:S and CS:GO?

Comment (Kefta) [inline gamemode/modules/fadmin/fadmin/changelevel/sv_init.lua]: Oops, I meant to search if it existed in cstrike or not, not csgo

**Comment by Kefta** [comment]
@bo98 Can you explain further? I don't understand

**Comment by Kefta** [comment]
@Bo98 Done, thank you for your help.

**Comment by Kefta** [comment]
Thanks

**Comment by FPtje** [comment]
Wait why was this made into the 2.7.0 branch?

**Comment by Kefta** [comment]
I swear I made it into the main

**Comment by Kefta** [comment]
@FPtje At the very least, it adds incentive for people to update!

**Pull request merged**